realm/src/main/java/io/realm/RealmList.java

```diff
@@ -66,11 +67,25 @@ public E set(int location, E object) {
     }
 
     /**
-     * Moves an object from one position to another
-     * @param oldPos
-     * @param newPos
+     * Moves an object from one position to another, while maintaining a fixed sized list.
+     * RealmObjects will be shifted so no null values are introduced.
+     *
+     * @param oldPos Index of RealmObject to move.
+     * @param newPos Target position. If newPos < oldPos the object at the location will be shifted
+     *               to the right. If oldPos < newPos, indexes > oldPos will be shifted once to the
+     *               left.
+     *
+     * @throws java.lang.IndexOutOfBoundsException if positions are outside [0, size()[.
      */
     public void move(int oldPos, int newPos) {
+        if (oldPos < 0 || newPos < 0) {
```

Comment:
All other error handling is done at the C++ jni level. So should it be for RealmList also.